web/src/hooks/useRobotAccounts.ts

```diff
@@ -182,3 +188,248 @@ interface createNewRobotForNamespaceParams {
   selectedTeams: IRobotTeam[];
   robotDefaultPerm: string;
 }
+
+export function useRobotPermissions({orgName, robName, onSuccess, onError}) {
+  const [namespace, setNamespace] = useState(orgName);
+  const [robotName, setRobotName] = useState(robName);
+  const [page, setPage] = useState(1);
+  const [perPage, setPerPage] = useState(10);
+
+  const {usernames} = useOrganizations();
+  const isUser = usernames.includes(orgName);
+
+  const {
+    data: robotPermissions,
+    isLoading: loading,
+    error,
+  } = useQuery(
+    ['Namespace', namespace, 'robot', robotName, 'permissions'],
```

Comment:
any reason why we need all these query keys?

Reply:
That's the query key for robot perms. I invalidate them [here](https://github.com/quay/quay/pull/2266/files#diff-ae5025f0fc8d98c584284426ef58c202d48f01937a65eff40694fc90ef01728aR229) and [here](https://github.com/quay/quay/pull/2266/files#diff-ae5025f0fc8d98c584284426ef58c202d48f01937a65eff40694fc90ef01728aR251) where repo perms are deleted/updated.

Reply:
right, my thinking was that having too many query key strings seems confusing (eg:`'Namespace'` seems redundant when you already have `'robots'` along with other variables making it unique enough). From what I understand, query key uniquely identifies a given query and refetches cache only when one of its variable changes or if you use `queryClient.invalidateQueries([<query-key])`.  So we should be okay using fewer keys as long as it's unique. wdyt?

Reply:
Different namespaces can have the same robot name. So, I am not sure if I can shorten it more. What do you have in mind?

Reply:
(discussed offline)